x-pack/plugins/osquery/server/plugin.ts

```diff
@@ -135,6 +136,9 @@ export class OsqueryPlugin implements Plugin<OsqueryPluginSetup, OsqueryPluginSt
         await this.initialize(core, dataViewsService);
       }
 
+      // Upgrade integration into 1.6.0 and rollover if found 'generic' dataset - we do not want to wait for it
+      upgradeIntegration({ packageInfo, client, esClient, logger: this.logger });
```

Comment:
Suggestion: as the upgrade is conditional, could this be renamed to something like `upgradeIntegrationIfNeeded` or `checkAndUpgradeIntegrationIfNeeded`